azure-iot-hub/azure/iot/hub/iothub_amqp_client.py

```diff
@@ -70,20 +113,24 @@ def send_message_to_device(self, device_id, message, app_props):
         message = uamqp.Message(
             msg_content, properties=msg_props, application_properties=app_properties
         )
-        self.amqp_client.queue_message(message)
-        results = self.amqp_client.send_all_messages(close_on_done=False)
+        self.amqp_message_send_client.queue_message(message)
+        results = self.amqp_message_send_client.send_all_messages(close_on_done=False)
         if uamqp.constants.MessageState.SendFailed in results:
             raise Exception("C2D message send failure")
 
 
 class IoTHubAmqpClientSharedAccessKeyAuth(IoTHubAmqpClientBase):
     def __init__(self, hostname, shared_access_key_name, shared_access_key):
-        self.endpoint = self._build_amqp_endpoint(
-            hostname, shared_access_key_name, shared_access_key
+        super(IoTHubAmqpClientSharedAccessKeyAuth, self).__init__()
+        self.hostname = hostname
+        self.shared_access_key_name = shared_access_key_name
+        self.shared_access_key = shared_access_key
+
+    def _get_target(self, operation):
+        endpoint = self._build_amqp_endpoint(
+            self.hostname, self.shared_access_key_name, self.shared_access_key
         )
-        operation = "/messages/devicebound"
-        target = "amqps://" + self.endpoint + operation
-        self.amqp_client = uamqp.SendClient(target)
+        return "amqps://" + endpoint + operation
 
     def _generate_auth_token(self, uri, sas_name, sas_value):
```

Comment:
Another thing that @cartertinney pointed out is that we could reduce code duplication by leveraging the [SasToken class](https://github.com/Azure/azure-iot-sdk-python/blob/master/azure-iot-hub/azure/iot/hub/sastoken.py)